diff --git a/lib/hot.js b/lib/hot.js
index 88bb139dced393fffa91b514d340bba818967ff7..d0331137d11ebe54b5f86dfdc8fa015e49ac50fa 100644
--- a/lib/hot.js
+++ b/lib/hot.js
@@ -115,7 +115,7 @@ class ChangeTracker {
         return promise;
     }
     async fetchEntities(meta, ids) {
-        let entities = await this.em.query(`SELECT * FROM ${this.escape(meta.tableName)} WHERE id = ANY($1::text[])`, [ids]);
+        let entities = await this.em.query(`SELECT * FROM ${this.escape(meta.tableName)} WHERE id = ANY($1)`, [ids]);
         // Here we transform the row object returned by the driver to its
         // JSON variant in such a way, that `driver.query('UPDATE entity SET field = $1', [json.field])`
         // would be always correctly handled.
diff --git a/src/hot.ts b/src/hot.ts
index 36fe84e1e1f4f0bda3185cdd914d4f2fdb0b7bbe..e63de109640ba55be6458fa81cadf3e1ef04dbb7 100644
--- a/src/hot.ts
+++ b/src/hot.ts
@@ -192,7 +192,7 @@ export class ChangeTracker {
 
     private async fetchEntities(meta: EntityMetadata, ids: string[]): Promise<Entity[]> {
         let entities = await this.em.query(
-            `SELECT * FROM ${this.escape(meta.tableName)} WHERE id = ANY($1::text[])`,
+            `SELECT * FROM ${this.escape(meta.tableName)} WHERE id = ANY($1)`,
             [ids]
         )
 
